### ISSUE TITLE
Why group by returns duplicate keys?

### ISSUE THREAD

**Issue by rainnnnny** [betodealmeida/shillelagh#368]
Hello again,  seems the group work is done from batch to batch,
I think this is done inside apsw, I'm just confused and wonder if it can be controled in shillelagh:
![image](https://github.com/betodealmeida/shillelagh/assets/15228876/168bd3b9-b1cc-4add-9788-9766050865fb)

in my adapter:
![image](https://github.com/betodealmeida/shillelagh/assets/15228876/6e80f796-d5d5-46c1-b8d3-bd83cfbfcbf3)

**Comment by betodealmeida** [comment]
@rainnnnny can you give some more details? Are you writing a mongo adapter?

**Comment by rainnnnny** [comment]
Sorry I had another issue right before this, so I omitted some info.

 I was writing a custom adapter of graphql based on [cancan101/graphql-db-api](https://github.com/cancan101/graphql-db-api), and I just changed the code of accessing graphql api of `graphql-db-api`, so I guess using the original `graphql-db-api` may reproduce, basicly the problem is that the data rows inside adapter are alright but the final results of `group by` is wrong.
 
Now I changed my implemention from `shillelagh + graphql-db-api` to `sqlglot + duckdb` and got rid of apsw which I considered not really controllable. So I'm sry the old code is no longer convenient to run again for you. Close this issue if you wish.
